openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionBodyGenerator.java

```diff
@@ -200,31 +212,40 @@ private void handleParameterSchemaInOperation(Map.Entry<PathItem.HttpMethod, Ope
         }
 
         if (operation.getValue().getParameters() != null) {
-            List<Parameter> parameters = operation.getValue().getParameters();
-            List<Parameter> queryParameters = new ArrayList<>();
-            List<Parameter> headerParameters = new ArrayList<>();
-            for (Parameter parameter: parameters) {
-                if (parameter.getIn().trim().equals("query")) {
-                    queryParameters.add(parameter);
-                } else if (parameter.getIn().trim().equals("header")) {
-                    headerParameters.add(parameter);
-                }
+            handleOperationWithParameters(operation, statementsList, queryApiKeyNameList, headerApiKeyNameList);
+        } else {
+            handleOperationWithoutParameters(statementsList, queryApiKeyNameList, headerApiKeyNameList);
+        }
```

Comment:
It seems that we have redundant lines of code in `handleOperationWithParameters`,  `handleOperationWithoutParameters`, `addUpdatedPathAndHeaders` and `addApiKeysToPathAndHeaders` functions. Shall we try to have a common approach as below ? 
```suggestion
        List<Parameter> queryParameters = new ArrayList<>();
        List<Parameter> headerParameters = new ArrayList<>();
        if (operation.getValue().getParameters() != null) {
            List<Parameter> parameters = operation.getValue().getParameters();
            for (Parameter parameter: parameters) {
                if (parameter.getIn().trim().equals(QUERY)) {
                    queryParameters.add(parameter);
                } else if (parameter.getIn().trim().equals(HEADER)) {
                    headerParameters.add(parameter);
                }
            }
        }

        handleOperationWithParameters(queryParameters, headerParameters, statementsList, queryApiKeyNameList, headerApiKeyNameList);
```